### PR TITLE
Refactor streams-bootstrap cleanup jobs as individual HelmApp

### PR DESCRIPTION
**Pull request by disrupted**
Depends on #397 

extracted the cleanup job needed for streams-bootstrap producer and streams apps to its own HelmApp nested inside the parent component

- [x] Create docs for `StreamsBootstrap` base
- [x] Update KPOps component inheritance graph
- [x] Update docs gen for multiple component inheritance (#406)

### CODE REVIEW

**Comment by sujuka99** [comment]
Please document the breaking changes in the migration guide

File: docs/docs/resources/pipeline-components/dependencies/defaults_pipeline_component_dependencies.yaml [hunk 1]
```diff
@@ -2,8 +2,10 @@ helm-app.yaml:
 - app-helm-app.yaml
 - repo_config-helm-app.yaml
 kafka-app.yaml:
+- prefix.yaml
```
Comment (disrupted) [inline docs/docs/resources/pipeline-components/dependencies/defaults_pipeline_component_dependencies.yaml:5]: weird that those weren't there before 😛 nobody noticed

**Review by sujuka99 — APPROVED** [review]

**Review by raminqaf — APPROVED** [review]

File: docs/docs/user/migration-guide/v2-v3.md [hunk 2]
```diff
@@ -40,6 +40,22 @@ All Helm-specific parts of the built-in [`KubernetesApp`](../core-concepts/compo
       ...
 ```
 
+## [Create StreamsBootstrap component & refactor cleanup jobs as individual HelmApp](https://github.com/bakdata/kpops/pull/398)
+
+Previously the default `KafkaApp` component configured the [streams-bootstrap](https://bakdata.github.io/streams-bootstrap/) Helm Charts. Now, this component is no longer tied to Helm (or Kubernetes). Instead, there is a new `StreamsBootstrap` component that configures the Helm Chart repository for the components that use it, e.g. `StreamsApp` and `ProducerApp`. If you are using non-default values for the Helm Chart repository or version, it has to be updated as shown below.
```
Comment (philipp94831) [inline docs/docs/user/migration-guide/v2-v3.md:45]: What is the point of having KafkaApp then?

Comment (disrupted) [reply docs/docs/user/migration-guide/v2-v3.md:45]: implement the logic related to Kafka, i.e. creation of output topics

**Pull request merged**